### PR TITLE
Include tag inheritance example in contracts docs

### PR DESCRIPTION
**Pull request by david-castaneda**
Show example of tag inheritance and add missing contracts error codes

### CODE REVIEW

File: studio-docs/source/contracts.mdx [hunk 1]
```diff
@@ -91,27 +91,30 @@ type Query {
...
 
-Whenever your source variant's [API schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
+Whenever your source variant's [Supergraph schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
 
 #### Tagging rules
 
 * You can apply tags to the following in your schema:
     * Fields of object types (as shown above)
     * Definitions of object, interface, and union types
+    * Tags on an object or interface are inherited by all of its fields (as shown above)
```
Comment (zionts) [inline studio-docs/source/contracts.mdx:117]: What happens when a type is _extended_ by another subgraph and fields are added, and the other subgraph doesn't tag the type?

Comment (david-castaneda) [reply studio-docs/source/contracts.mdx:117]: So tags get merged across subgraphs and extended types. So if `A` has tag `public` in subgraph A and extension of `A` has tag `private` in subgraph B, in the supergraph schema all fields on `A` will contain both the `public` and `partner` tag.

Definitely could be more clear on the behavior:
```
Tags applied to objects and interfaces (as well as tags applied to extension types) are inherited by all fields of that type.
```

File: studio-docs/source/contracts.mdx [hunk 2]
```diff
@@ -91,27 +91,30 @@ type Query {
   topProducts: [Product!]!
 }
 
-type Product @key(fields: "upc") {
+# All fields on the "Product" object inherit the "partner" tag so we can avoid tagging them explicitly
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:94]: All fields of the Product object type automatically inherit...

explicitly -> individually (either works fine if you'd rather keep 😄)

File: studio-docs/source/contracts.mdx [hunk 3]
```diff
@@ -91,27 +91,30 @@ type Query {
...
+type Product @key(fields: "upc") @tag(name: "partner") {
   upc: ID!
   name: String!
   description: String!
-  # Tagged fields
+  # The internalId, and percentageMatch fields would also inherit the "partner" tag, so they need
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:99]: cut first comma

would also -> also

File: studio-docs/source/contracts.mdx [hunk 4]
```diff
@@ -91,27 +91,30 @@ type Query {
...
+  # the "private" and "experimental" tag (and an exclude filter) to keep being inaccessible
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:100]: and an -> along with an

keep being -> remain

File: studio-docs/source/contracts.mdx [hunk 5]
```diff
@@ -91,27 +91,30 @@ type Query {
...
   internalId: ID! @tag(name: "private")
   percentageMatch: Float! @tag(name: "experimental")
 }
 ```
 
-This schema applies the `@tag` directive to two object fields: `internalId` and `percentageMatch`.
+This schema applies the `@tag` directive to one object: `Product`, and two object fields: `internalId` and `percentageMatch`.
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:106]: one object (`Product`) and two object fields (`internalId` and `percentageMatch`).

File: studio-docs/source/contracts.mdx [hunk 6]
```diff
@@ -91,27 +91,30 @@ type Query {
...
 }
 ```
 
-This schema applies the `@tag` directive to two object fields: `internalId` and `percentageMatch`.
+This schema applies the `@tag` directive to one object: `Product`, and two object fields: `internalId` and `percentageMatch`. 
 
 Each tag has a string `name`. You tag types and fields with the _same_ `name` if they should be included or excluded as a group by a particular contract.
 
-Whenever your source variant's [API schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
+Whenever your source variant's [Supergraph schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:110]: lowercase supergraph

File: studio-docs/source/contracts.mdx [hunk 7]
```diff
@@ -206,17 +209,21 @@ Whenever Apollo Studio attempts to create or update your contract schema, it mig
 
 | Error | Description |
 |-------|-------------|
+| `ADD_DIRECTIVE_DEFINITIONS_IF_NOT_PRESENT` | An error occurred adding directive definitions for `@tag`, `@inaccessible`, and core directive usages. |
+| `DIRECTIVE_DEFINITION_LOCATION_AUGMENTING` | An error occured augmenting the directive definition for `@tag` to support `OBJECT`, `FIELD_DEFINITION`, `INTERFACE` and `UNION`. |
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:213]: serial comma after `INTERFACE`

File: studio-docs/source/contracts.mdx [hunk 8]
```diff
@@ -206,17 +209,21 @@ Whenever Apollo Studio attempts to create or update your contract schema, it mig
...
 | `PARTIAL_INTERFACE_MASKING` | An interface field's return type was excluded, and an error occurred while excluding that interface field. |
 | `SCHEMA_RETRIEVAL` | Studio encountered an error while retrieving the source variant's schema. It might not yet have a valid composed schema. |
-| `TAG_MATCHING` | Studio encountered an error while determining which tags to include and exclude. The contract might be attempting to include and exclude the same tag. |
+| `TAG_INHERITING` | Studio encountered an error while attempting to add parent tags to fields.  |
+| `TAG_MATCHING` | Studio encountered an error determining which types and fields should be inaccessible based on their tags. |
 | `TO_FILTER_SCHEMA` | Studio failed to generate and return a contract supergraph schema for an unknown reason. |
 | `TO_API_SCHEMA` | Studio encountered an error while attempting to generate an API schema from the contract variant's supergraph schema. |
 | `UNKNOWN` | An unknown error occurred. |
+| `VERSION_CHECK` | The Federation version used is not supported by contracts. |
```
Comment (StephenBarlow) [inline studio-docs/source/contracts.mdx:226]: At this point the table is _almost_ alphabetically sorted. Would you mind taking it the rest of the way in this PR? 😄

**Review by StephenBarlow — APPROVED** [review]
Approved with small suggestions!

**Pull request merged**